packages/tfchain_client/src/client.ts

```diff
@@ -120,6 +121,87 @@ class QueryClient {
     await this.connect();
     return await func.apply(this, args);
   }
+
+  /**
+   * Checks if the given section exists within the API events.
+   *
+   * @param {string} section - The section to check within the API events.
+   * @returns {boolean} - True if the section exists within the API events, false otherwise.
+   */
+  private checkSection(section: string): boolean {
+    const sections = Object.keys(this.api.events);
+    if (sections.includes(section)) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Checks if the given method exists within the specified section of the API events.
+   *
+   * @param {string} section - The section of the API events to check.
+   * @param {string} method - The method to check within the section.
+   * @returns {boolean} - True if the method exists within the section, false otherwise.
+   */
+  private checkMethod(section, method) {
+    const methods = Object.keys(this.api.events[section]);
+    if (methods.includes(method)) {
+      return true;
+    } else return false;
+  }
+
+  /**
+   * Listens for a specific event on the chain and resolves when the event matches the specified conditions.
+   *
+   * @param {string} section - The section of the event to listen for.
+   * @param {string} method - The method of the event to listen for.
+   * @param {validatorFunctionType} validator - The validator function to validate the event data.
+   * @param {number} time - The timeout value in milliseconds. Default is 2 minutes.
+   * @returns {Promise<object>} - A promise that resolves with the event data when the event matches the conditions.
+   * @throws {Error} - If the section or method is not defined on the chain.
+   * @rejects  If no response is received within the given time or if an error occurs during validation.
+   */
+  async listenForEvent<T>(
+    section: string,
+    method: string,
+    validator: validatorFunctionType,
+    time = 120000,
```

Comment:
can you convert this to minutes?
```suggestion
    timeoutInMinutes = 120000,
```